frontend/src/metabase/containers/NewItemMenu/NewItemMenu.tsx

```diff
@@ -1,36 +1,56 @@
 import { connect } from "react-redux";
+import { push } from "react-router-redux";
 import _ from "underscore";
-import { closeNavbar } from "metabase/redux/app";
+
 import NewItemMenu from "metabase/components/NewItemMenu";
+
 import Databases from "metabase/entities/databases";
+import Search from "metabase/entities/search";
+
+import { closeNavbar } from "metabase/redux/app";
 import {
   getHasDataAccess,
   getHasDatabaseWithJsonEngine,
   getHasNativeWrite,
+  getHasDatabaseWithActionsEnabled,
 } from "metabase/selectors/data";
+
 import { Database } from "metabase-types/api";
 import { State } from "metabase-types/store";
 
 interface MenuDatabaseProps {
   databases?: Database[];
+
+  // Search results (untyped yet)
+  models?: unknown[];
 }
 
 const mapStateToProps = (
   state: State,
-  { databases = [] }: MenuDatabaseProps,
+  { databases = [], models = [] }: MenuDatabaseProps,
 ) => ({
+  hasModels: models.length > 0,
   hasDataAccess: getHasDataAccess(databases),
   hasNativeWrite: getHasNativeWrite(databases),
   hasDatabaseWithJsonEngine: getHasDatabaseWithJsonEngine(databases),
+  hasDatabaseWithActionsEnabled: getHasDatabaseWithActionsEnabled(databases),
```

Comment:
Yeah, that'd be nicer for a view component. I'm a bit on the fence here because I'm not sure I like the fact that we're going to spread these checks across the view and container components. In that way, something is going to be derived in a container and rest — in the view, so it could become confusing